spec/routine_spec.rb

```diff
@@ -116,4 +116,40 @@ def exec
     end
   end
 
+  it 'does not explode on a transaction retry' do
+    # To get the transaction to retry (and to simulate what happens in production),
+    # we need to raise an exception the first time through after the event has been
+    # set in the routine outputs.  I picked a call to BL to fake b/c it doesn't impact
+    # the outcome.  On the first call, the BL call raises an exception that triggers
+    # transaction_retry; on the second it doesn't.
+    #
+    stub_const 'RaiseTransactionIsolationConflict', Class.new
+    RaiseTransactionIsolationConflict.class_eval {
+      lev_routine
+      def exec
+        @times_called ||= 1
+        raise(::ActiveRecord::TransactionIsolationConflict, 'hi') if @times_called == 1
+        @times_called += 1
+      end
+    }
+
+    stub_const 'MainRoutine', Class.new
+    MainRoutine.class_eval {
+      lev_routine
+      def exec
+        outputs[:test] = 1
+        RaiseTransactionIsolationConflict.call
```

Comment:
Hmm.... I think what we need is for the nested routine to set outputs and to have the parent routine translate those verbatim.  This will mean that the second run will append the outputs instead of just setting them.  And then the expectation needs to expect that there isn't an array of outputs, just one output.  I will write something...